### PR TITLE
Implement Supervisor::SetupDInput and callbacks

### PR DESCRIPTION
**Pull request by Renzo904**
- Implemented Supervisor::SetupDInput
- Implemented DInput callbacks "controllerCallback" and "EnumGameControllersCb"
- Added dinput8.lib to the linker

### CODE REVIEW

File: src/Supervisor.cpp [hunk 1]
```diff
@@ -867,3 +954,47 @@ u16 GetInput(void)
 
     return GetControllerInput(buttons);
 }
+
+#pragma optimize("s", on)
+#pragma var_order(diprange, result)
+BOOL CALLBACK controllerCallback(LPCDIDEVICEOBJECTINSTANCEA lpddoi, LPVOID pvRef)
```
Comment (roblabla) [inline src/Supervisor.cpp:960]: Please use PascalCase for functions (capitalize the first letter)

File: src/Supervisor.cpp [hunk 2]
```diff
@@ -867,3 +954,47 @@ u16 GetInput(void)
...
+
+{
+    HRESULT result;
+    DIPROPRANGE diprange;
+    result = (HRESULT)pvRef; // ehm, ok?
```
Comment (roblabla) [inline src/Supervisor.cpp:965]: Maybe drop the cast, retype `result` to LPVOID, and rename it to something like pvRefBackup or something? IMO it'd make a bit more sense.

**Review by roblabla — COMMENTED** [review]
Small nitpicks, looks very solid otherwise!

File: src/Supervisor.cpp (hunk 2 above)
Comment (Renzo904) [reply src/Supervisor.cpp]: Makes sense, ghidra's output led me to believe at first that it was the result of `g_Supervisor.controller->SetProperty`

**Comment by roblabla** [comment]
Thank you for your contribution!

**Pull request merged**